### PR TITLE
feat: Support multiple models in S3 batch export 

### PR DESCRIPTION
**Pull request by tomasfarias**
## Problem

Users wish to export person information to allow identifying unique persons in their event batch exports.

## Changes

This commit introduces a new `batch_export_model` input for all batch export destinations. This input is of type `BatchExportModel` which can be used to indicate which model a batch export is supposed to target. This opens the door to creating "persons" model exports, and eventually extending this further with more changes.

The change was done in a backwards compatible way, so `batch_export_schema` was not removed, and existing export configurations should continue to work (as evidenced by unit tests passing without changes).

However, moving forward all batch exports will be created with `batch_export_schema` set to `None` and `batch_export_model` defined in its place.

After updating existing exports, `batch_export_schema` and any other code associated with backwards compatibility can be deleted.

👉 _Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review._

## How did you test this code?

Updated all S3 unit tests to guarantee person batch export support. Follow-up PRs will work on other destinations.

This required generating person data, and checking it was successfully exported. To do this, new utilities were added under `temporal/tests/utils/persons.py`, they work similar to event utilities in `temporal/tests/utils/event.py`

Also, a bug was fixed in the `test_s3_export_workflow_with_minio_bucket_and_custom_key_prefix` test as it wasn't properly using custom keys.

### CODE REVIEW

**Comment by tomasfarias** [comment]
Hmm not sure why the hogql_query tests are failing, don't see any changes of mine around that part of the codebase. I'll investigate tomorrow.

**Review by EDsCODE — APPROVED** [review]
lgtm, nice pattern for extending beyond persons

File: docker-compose.dev.yml [hunk 1]
```diff
@@ -48,8 +48,6 @@ services:
             - ./docker/clickhouse/docker-entrypoint-initdb.d:/docker-entrypoint-initdb.d
             - ./docker/clickhouse/config.xml:/etc/clickhouse-server/config.xml
             - ./docker/clickhouse/users-dev.xml:/etc/clickhouse-server/users.xml
-        extra_hosts:
-            - 'host.docker.internal:host-gateway'
```
Comment (tomasfarias) [inline docker-compose.dev.yml:52]: Removed this as locally was failing by builds. Shouldn't have been pushed though.

**Comment by tomasfarias** [comment]
Hey @fuziontech! You got tagged for review as I rebased this PR and commits from the previous PR you've reviewed popped up in this PR's GitHub history.

Feel free to ignore, and apologies for the spam. Unless you are curious, then obviously comments are welcome!

**Pull request merged**